### ISSUE TITLE
`extra-executables` in cabal.project

### ISSUE THREAD

**Issue by fgaz** [haskell/cabal#9230]
**Describe the feature request**
A new `extra-executables` field that specifies executables to be provided during development, such as linters and formatters. These should be solved independently from each other and from the main project.

The field should **not** contain executables that are needed to _build_ the project. Those go in `build-tool-depends`, in the appropriate .cabal file

**Additional context**
Originally #7865 and https://github.com/haskell/cabal/issues/6952#issuecomment-971440004

**Comment by mpickering** [comment]
I think this is a bit underspecified. 

How is this field used? What commands would invoke these executables to be built, and how would they end up in the PATH?

**Comment by michaelpj** [comment]
I tend to agree. If we had a concept of a cabal-provided dev-shell, and the field was `dev-tools` or something, then it would make sense that when you go into the dev shell you get the dev tools on the path. But just saying they're "extra" is indeed rather unclear.

**Comment by andreabedini** [comment]
I would find this confusing since we already have `extra-packages` which are solved _within_ the plan.

TBH I am -1 on adding dev environment management features to cabal.

If those executables have to solved independently; this would be equivalent to using `cabal install` to install them.
